src/trees/FunctionTreeVector.h

```diff
@@ -1,38 +1,53 @@
 #pragma once
 
+#include <tuple>
 #include <vector>
 
 #include "FunctionTree.h"
 
 namespace mrcpp {
 
-template<int D>
-class FunctionTreeVector {
-public:
-    FunctionTreeVector() { }
-    virtual ~FunctionTreeVector() { }
-
-    FunctionTreeVector(const FunctionTreeVector<D> &vec);
-    FunctionTreeVector& operator=(const FunctionTreeVector<D> &vec);
-
-    int size() const { return this->funcs.size(); }
-    void clear(bool dealloc = false);
-    int sumNodes() const;
-
-    void push_back(double c, FunctionTree<D> *f);
-    void push_back(FunctionTree<D> *f);
-    void push_back(FunctionTreeVector<D> &vec);
+template <int D>
+using CoefsFunctionTree = std::tuple<double, FunctionTree<D> *>;
+
+template <int D>
+using FunctionTreeVector = std::vector<CoefsFunctionTree<D>>;
+
+template <int D>
+void clear(FunctionTreeVector<D> & fs, bool dealloc = false) {
+   if (dealloc) {
+     for (auto & t : fs) {
+       auto f = std::get<1>(t);
+       if (f != nullptr) delete f;
+     }
+   }
+   fs.clear();
+}
 
-    double getCoef(int i) const;
-    FunctionTree<D> &getFunc(int i);
-    const FunctionTree<D> &getFunc(int i) const;
+template <int D>
+int sumNodes(const FunctionTreeVector<D> & fs) {
```

Comment:
Please use `sum_nodes()` for free functions, and `sumNodes()` for class functions

Reply:
Is this convention documented? :)

Reply:
BTW, just noticed `sumNodes` (or the prospective `sum_nodes`) is not used anywhere within MRCPP. Is it used in MRChem or can it be removed altogether?

Reply:
Done.

Reply:
Yes, it's used in MRChem and is sort of part of the API

Reply:
And the naming conventions will be suddenly documented :wink: